IMC_R/Modeling_Jeff/Intro to Logistic Regression.md

```diff
@@ -0,0 +1,99 @@
+
+The final step would be to create the confusion matrix to determine the true/false postive and the true/false negatives values of the patients' diagnosis, the accuracy level of the logistic regression model itself, and the confidence interval of the true accuracely level of how the logistic regression model performs.
+
+```r
+#Creating the confusion matrix that will also give the accuracy level
+confusionMatrix(as.factor(test$target),as.factor(prediction))
+```
+
+![](https://i.paste.pics/cd29f1cbf3e96a02d6ddc44e958bae00.png)
+
+This is the numerical information regarding the confusion matrix of our logistic regression model. We can see that that the accuracy level is 80.22% (Accuracy : 0.8022), which means that our model has a high accuracy level of predicting the proper diagnosis of heart disease patients. The 95% confidence interval is between 70.55% to 87.94% (95% CI : (0.7055, 0.8784)) which means that overall, model is fairly accurate since the range of accuracy is much higher than 50%. In terms of the true/false postive and the true/false negative values of each patient's diagnosis, we can look at the true positive rate by looking at the Pos Pred Value entry, which is 0.8049 (80.49%). We can also look at the false positive rate by doing 1 - (Pos Pred Value) which would give us 0.1951 (19.51%). This means that 80.49% of patients were correctly diagnosed to have heart disease and 19.51% of patients were incorrectly diagnosed with heart disease (true postive diagnosed patients = 40, false positive diagnosed patients = 8). As for the true negative rate, we can use the Neg Pred Value entry, which is 0.8000 (80%). We can also look for the false negative rate by doing 1 - (Neg Pred Value) which would give us 0.2000 (20%). This means that 80% of patients were correctly misdiagnosed for heart disease and 20% of patients were incorrectly misdiagnosed for heart disease (true negative misdiagnosed patients = 33, false negative misdiagnosed patients = 10). So if we were to add the total amount of patients that actually have heart disease, it would be 50 people in total (40 + 10 = 50) and if we were to add the total number of people that don't have heart disease, it would be 41 in total (33 + 8 = 41). We can see that the majority of pateints do have heart disease, but there is a moral consequence of clasification error. A false positive for diagnosis would mean that the doctor would waste time and resources on a patient that they believe has heart disease, when they're actually fine. A false negative would mean that the doctor would ignore patients that they believe were misdiagnosed, but they are in dire need of help and cold die of a heart attack at any minute without aid. Luckily it was only a small percent of patients that fall into both groups, but it's still a risky error a doctor can make on the livelihood of their patients.
```

Comment:
if you want the result to be reproducible, considering putting "set.seed(123)" or some arbitrary number before the random 70/30 split